### ISSUE TITLE
make one generic fault implementation with start and clearing times

### ISSUE THREAD

**Issue by SabineAuer** [JuliaEnergy/PowerDynamics.jl#87]
see PowerPerturabtion implementation

**Comment by luap-pik** [comment]
@SabineAuer Could you clarify what you mean here? Do you want a dispatch that automatically determines if a fault should hit a line/node based on the passed arguments? Also, there are more parameters needed, e.g. the fault resistance.

**Comment by SabineAuer** [comment]
I meant something else. Sorry for the very short explanation. Doing lab measurements at Tecnalia we realized we need for every fault to be able to have a starting time and end time different from the simulation start and end time. Just like we changed the `simulate`-function in this respect only for the PowerPerturbation, see [here](https://github.com/JuliaEnergy/PowerDynamics.jl/blob/dfcd8fb8bb6cc270fcbd846419fd1cfd33d44d51/src/simulations/PowerPerturbation.jl#L48).

**Comment by luap-pik** [comment]
Ah ok that makes sense. We have the same for the `NodeShortCircuit`.

Btw., is there a reason why [here](https://github.com/JuliaEnergy/PowerDynamics.jl/blob/dfcd8fb8bb6cc270fcbd846419fd1cfd33d44d51/src/simulations/PowerPerturbation.jl#L59) you don't setp to the fault point exactly but only +-dt ?

**Comment by SabineAuer** [comment]
@luap-pik : In the `PowerPerturbation` Case we had to add one "trick" that is specifically necessary for changing the parameters of the algebraic constrained nodes. We had to tell the integrator to step specifically into time steps shortly before and after the discontinuity (see [here](https://github.com/JuliaEnergy/PowerDynamics.jl/blob/dfcd8fb8bb6cc270fcbd846419fd1cfd33d44d51/src/simulations/PowerPerturbation.jl#L64) in power because for algebraic nodes the integrator will otherwise step over the discontinuity and the simulation results turn out wrong. Did I explain this understandably?
